transformer_engine/common/include/transformer_engine/cast.h

```diff
@@ -42,6 +42,25 @@ extern "C" {
  *  of the output tensor should be set to 0.
  */
 
+/*! \brief Find amax value of input tensor for FP8 quantier.
+ *
+ *  \param[in]     input            Input tensor to be cast.
+ *  \param[in,out] output           Output FP8 tensor.
+ *  \param[in]     stream           CUDA stream used for the operation.
+ */
+void nvte_compute_amax(const NVTETensor input, NVTETensor output, cudaStream_t stream);
+
+/*! \brief Find scale value of input tensor for FP8 quantier from amax.
+ *         This function is only for per-tensor Current Scaling.
+ *         For subtensor scaling, find amax & quantize should be fused into one kernel.
+ *         This function is called after nvte_compute_amax. Amax reduction is optional
+ *         depending on model parallel configuration.
+ *
+ *  \param[in,out] output           Output FP8 tensor (just finished nvte_compute_amax).
+ *  \param[in]     stream           CUDA stream used for the operation.
+ */
+void nvte_compute_scale_from_amax(const NVTETensor output, cudaStream_t stream);
```

Comment:
```suggestion
void nvte_compute_scale_from_amax(const NVTETensor output, const NVTEQuantizationParams params, cudaStream_t stream);
```
We could make the NVTEQuantizationParams `void*` just like the `NVTETensor` to be able to change it without changing the API here if needed.

Reply:
This can be a good idea for this PR. 

If we think about @kwyss-nvidia 's PR about subtensor scaling, in which compute_amax, compute_scale and quantize are actually in one kernel, we will inevitably change the API of quantization. 

But maybe it's the correct way forward despite it can be a little painful.

Reply:
Done